AGENTS.md

```diff
@@ -0,0 +1,258 @@
+├── requirements.txt     # Python dependencies
+└── run_tests.sh         # Main test runner script
+```
+
+### Key Directories
+
+- **`features/`**: Contains all BDD feature files (`.feature`) and their step implementations. Each service has its own subdirectory. You should use different files files for different features, for example if you are testing an integration with a service, dedicate one file for that specific purpose. This helps organizing the scenarios and makes it easier to understand the purpose of each test.
+- **`features/steps/`**: Python implementations of Gherkin step definitions.
+- **`config/`**: Service configuration files used during testing.
+- **`mocks/`**: Mock implementations of external services (not part of the processing team repositories) to enable isolated testing. It is preferable to use mocks instead of adding dependencies that may need to run as containers or in the background.
+- **`test_data/`**: Sample data files used in test scenarios.
+- **`setup/`**: SQL scripts for database initialization and cleanup.
+
+## Best Development Practices
+
+As an agent, you should create a TODO list with the following sections:
+
+### Code Style and Quality
+
+1. **Python Code Style**
+   - Run `make code-style` before committing to check Python code style
+   - Use `make ruff` to run the Ruff linter
+   - Follow PEP 8 conventions
+   - Run `make docs-style` to check documentation strings
+
+2. **Shell Scripts**
+   - Run `make shellcheck` to validate shell scripts
+   - Use `#!/bin/bash -x` for debugging when needed
+
+3. **Type Checking**
+   - Install type libraries: `make install-type-libraries`
+   - Run type checks: `make type-checks`
+   - For strict checking: `make strict-type-checks`
+
+### BDD Test Development
+
+1. **Writing Feature Files**
+   - Use clear, descriptive scenario names
+   - Follow Given-When-Then structure
+   - Use appropriate tags (`@skip`, `@managed`, `@local`, etc.)
+   - Document scenarios with clear business language
+
+2. **Step Implementations**
+   - Reuse existing step definitions when possible
+   - Keep step implementations focused and testable
+   - Use the `context` object to share state between steps
+   - Handle cleanup in `after_scenario` hooks
+
+3. **Test Organization**
+   - Group related scenarios in feature files
+   - Use tags to categorize tests (`@database`, `@rest-api`, `@s3`, etc.)
+   - Update scenario lists: `make update-scenarios`
+
+### Keeping documentation up to date
+
+The documentation is generated from the scenario list in the `docs/scenarios_list.md` file, that is then rendered on the GitHub pages.
+To keep the documentation up to date, you need to run `make update-scenarios` after adding new scenarios or modifying existing ones.
+
+### Pre-Commit Checklist
+
+Before committing, run:
+```bash
+make before_commit
+```
+
+to ensure that the code style is correct and the documentation is up to date.
+
+## How to Run Locally
```

Comment:
Why we need this in a agents.md file? I understand this for a Readme for other users, but for an agent? What is the purpose?

Reply:
should be solved in https://github.com/RedHatInsights/insights-behavioral-spec/pull/725/commits/23f992d7e7a8c0988bbd5e54b6fd13f1e6320de3